common/sgx/tcbinfo.h

```diff
@@ -76,29 +122,96 @@ oe_result_t oe_verify_ecdsa256_signature(
     sgx_ecdsa256_signature_t* signature,
     oe_cert_chain_t* tcb_cert_chain);
 
+/*! \enum oe_qe_identity_id
+ *  \brief Quoting enclave identity id (V2 only)
+ */
+typedef enum _oe_qe_identity_id
+{
+    QE_IDENTITY_ID_QE,
+    QE_IDENTITY_ID_QVE
+} oe_qe_identity_id_t;
+
+/*! \struct oe_qe_tcb_level
+ *  \brief Quoting enclave identity TCB level.  Applies to V2 only.
+ */
+typedef struct _oe_qe_identity_info_tcb_level
+{
+    uint32_t isvsvn[1];
+    oe_tcb_level_status_t tcb_status;
+    oe_datetime_t tcb_date;
+
+    /*! Offset into the json QE Identity info where
+     * the advisoryIDs fields start.
+     */
+    size_t advisory_ids_offset;
+
+    //! Total size of all the advisoryIDs.
+    size_t advisory_ids_size;
+} oe_qe_identity_info_tcb_level_t;
+
+/*! \struct oe_parsed_qe_identity_info_t
+ *  \brief SGX Quoting Enclave Identity Info data structure.
+ */
 typedef struct _oe_parsed_qe_identity_info
 {
     uint32_t version;
     oe_datetime_t issue_date;
     oe_datetime_t next_update;
-    uint32_t miscselect;         // The MISCSELECT that must be set
-    uint32_t miscselect_mask;    // Mask of MISCSELECT to enforce
-    sgx_attributes_t attributes; // flags and xfrm (XSAVE-Feature Request Mask)
-    uint64_t attributes_flags_mask;   // mask for attributes.flags
-    uint64_t attributes_xfrm_mask;    // mask for attributes.xfrm
-    uint8_t mrsigner[OE_SHA256_SIZE]; // MRSIGNER of the enclave
-    uint16_t isvprodid;               // ISV assigned Product ID
-    uint16_t isvsvn;                  // ISV assigned SVN
+    uint32_t miscselect;         //! The MISCSELECT that must be set
+    uint32_t miscselect_mask;    //! Mask of MISCSELECT to enforce
+    sgx_attributes_t attributes; //! flags and xfrm (XSAVE-Feature Request Mask)
+    uint64_t attributes_flags_mask;   //! mask for attributes.flags
+    uint64_t attributes_xfrm_mask;    //! mask for attributes.xfrm
+    uint8_t mrsigner[OE_SHA256_SIZE]; //! MRSIGNER of the enclave
+    uint16_t isvprodid;               //! ISV assigned Product ID
+    uint16_t isvsvn;                  //! ISV assigned SVN
```

Comment:
>isvsvn [](start = 13, length = 6)

This becomes part of _oe_qe_identity_info_tcb_level in V2. Are we keeping to for V1 backward-compatibility? Is it in sync with tcb_level.isvsvn in V2 implementation?

Reply:
See the code in tcbinfo.c.

---
In reply to: [363009024](https://github.com/openenclave/openenclave/pull/2308#discussion_r363009024) [](ancestors = 363009024)

Reply:
Yes, V1 compatibility is preserved.  In V2 the isvsvn was moved to a tcb level section.  The code will keep the isvsvn field in this structure updated.  See line 1147 in tcbinfo.c